corporate/tests/test_stripe.py

```diff
@@ -7768,6 +7783,34 @@ def test_upgrade_user_to_fixed_price_plan_pay_by_invoice(self, *mocks: Mock) ->
         self.assertEqual(invoice.status, Invoice.PAID)
         self.assertEqual(fixed_price_plan_offer.status, CustomerPlanOffer.PROCESSED)
 
+        # Visit /billing
+        self.execute_remote_billing_authentication_flow(
+            hamlet, expect_tos=False, first_time_login=False
+        )
+        with (
+            time_machine.travel(self.now + timedelta(days=1), tick=False),
+            patch(
+                "corporate.lib.stripe.customer_has_credit_card_as_default_payment_method",
+                return_value=False,
+            ),
+            patch(
+                "stripe.Customer.retrieve",
+                return_value=Mock(id=stripe_customer_id, email=hamlet.delivery_email),
+            ),
+            patch("stripe.Invoice.retrieve", return_value=mock_invoice),
+        ):
+            response = self.client_get(
+                f"{self.billing_session.billing_base_url}/billing/", subdomain="selfhosting"
+            )
+        for substring in [
+            "Zulip Basic",
+            hamlet.delivery_email,
+            "Annual",
+            "This is a fixed-price plan",
+            "You will be contacted by Zulip Sales",
+        ]:
+            self.assert_in_response(substring, response)
```

Comment:
Added this to the tests since I think we might have caught this issue earlier if we'd checked visiting the billing page.